libr/core/cmd_search.c

```diff
@@ -258,7 +258,17 @@ static void cmd_search_bin(RCore *core, RInterval itv) {
 		if (plug) {
 			r_cons_printf ("0x%08" PFMT64x "  %s\n", from, plug->name);
 			if (plug->size) {
-				r_bin_load_io (core->bin, core->file->fd, 0, 0, 0, core->offset, plug->name, 4096);
+				RBinOptions opt = {
+					.pluginname = plug->name,
+					.offset = core->offset,
+					.baseaddr = 0,
+					.loadaddr = 0,
+					.sz = 4096,
+					.xtr_idx = 0,
+					.rawstr = core->bin->rawstr,
+					.fd = core->file->fd,
+				};
+				r_bin_open_io (core->bin, &opt);
```

Comment:
there shoulkd be no different between RBin.open and RBin.open_io, the last one should die, but in a diferent PR

Reply:
I think we should just have a `r_bin_open_buffer`. Bin should have nothing to do with IO. Everything should be done through IO and I think RBin shouldn't even depend on RIO.